src/Sentry/Internal/SentryScopeManager.cs

```diff
@@ -86,21 +90,21 @@ public void BindClient(ISentryClient client)
             _options?.DiagnosticLogger?.LogDebug("Binding a new client to the current scope.");
 
             var currentScopeAndClientStack = ScopeAndClientStack;
-            currentScopeAndClientStack = currentScopeAndClientStack.Pop(out var top);
-            currentScopeAndClientStack = currentScopeAndClientStack.Push(
-                new ValueTuple<Scope, ISentryClient>(top.Item1, client ?? DisabledHub.Instance));
-            ScopeAndClientStack = currentScopeAndClientStack;
+            var newScopeAndClientStack = new Stack<KeyValuePair<Scope, ISentryClient>>(currentScopeAndClientStack);
```

Comment:
`currentScopeAndClientStack` is a copy to a mutable type, I'm not sure it's safe to pass it to `Stack`'s constructor. I didn't read the code but assume it'll iterate over to copy the references and that'll throw if we mutate it.

Reply:
Not sure I understand. currentScopeAndClientStack is never modified. newScopeAndClientStack is a copy off currentScopeAndClientStack

Reply:
replied above